adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -1562,11 +1562,33 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     stringency: ValidationStringency = ValidationStringency.STRICT)(implicit s: DummyImplicit): AlignmentRecordRDD = LoadIndexedBam.time {
 
     val path = new Path(pathName)
-    // todo: can this method handle SAM and CRAM, or just BAM?
+
+    // If pathName is a single file or *.bam, append .bai to find all bam indices.
+    // Otherwise, pathName is a directory and the entire path must be searched
+    // for indices.
```

Comment:
Can you note:
```
// TODO: support BAMs where the index is located at s/.bam/.bai/g
```

These aren't common (`myFile.bam` + `myFile.bai` vs `myFile.bam.bai`) but you do see them in the wild occasionally.

Reply:
Maybe create an issue for that after this merges.

Reply:
Actually, if you do `new Path(pathName.toString.replace(".bam", "*.bai"))` here and make a change below (will drop a comment), I think you can handle both cases.